packages/task/src/browser/task-schema-updater.ts

```diff
@@ -67,6 +70,36 @@ export class TaskSchemaUpdater {
         }
     }
 
+    /**
+     * Adds given task schema to `taskConfigurationSchema` as `oneOf` subschema.
+     * Replaces existed subschema by given schema if the corrresponding `$id` properties are equal.
+     *
+     * Note: please provide `$id` property for subschema to have ability remove/replace it.
+     * @param schema subschema for adding to `taskConfigurationSchema`
+     */
+    addSubschema(schema: IJSONSchema): void {
+        const schemaId = schema.$id;
+        if (schemaId) {
+            this.removeSubschema(schemaId);
```

Comment:
How about extracting protected `doRemoveSchema` which does not do update, but returns boolean flag. Otherwise `addSubschema` can cause `update` twice.

Or change `customSchemas` type to `Map`.

Reply:
updated, thank you!